### PR TITLE
New documentation!

### PR DESCRIPTION
**Pull request by sihil**
This overhauls Riff-Raff's documentation:
 - Makes the Riff-Raff homepage display the documentation index
 - Restructure the docs so they all live in one place (the Riff-Raff sub project)
 - Change the index to be more user friendly
 - Adds documentation for the `riff-raff.yaml` file and the S3 bucket layout
 - Adds warnings to legacy docs that might still be of interest
 - Drops the paper thin Scala markdown wrapper in favour of using PegDown directly

### CODE REVIEW

File: riff-raff/public/docs/howto/configure-a-project.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+=====================
+
+To make a project deployable with riff-raff it needs:
+ 
+ - A [`riff-raff.yaml`](../reference/riff-raff.yaml.md) file that describes the deployment process
+ - A [`build.json`](../reference/build.json.md) file that details the CI build
+ - The two files above along with any assets uploaded into riff-raff's S3 artifact buckets
+
+Both the [sbt-riffraff-artifact plugin](https://github.com/guardian/sbt-riffraff-artifact) or
+[node-riffraff-artifact plugin](https://github.com/guardian/node-riffraff-artefact) will both create the `build.json`
```
Comment (philwills) [inline riff-raff/public/docs/howto/configure-a-project.md]: Repetition of 'both'.

File: riff-raff/public/docs/howto/configure-a-project.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+for you and help you to upload the files to the S3 buckets correctly.
+
+Using SBT
+---------
+
+ 1. Add [sbt-riffraff-artifact plugin](https://github.com/guardian/sbt-riffraff-artifact) to your `project/plugins.sbt`
```
Comment (philwills) [inline riff-raff/public/docs/howto/configure-a-project.md]: Should we just link to the sbt-riffraff-artifact docs? I'd worry about the two diverging.

File: riff-raff/public/docs/howto/configure-a-project.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+       the `riffRaffNotifyTeamcity` task
+    b. Otherwise, or if you are not sure, you should use the `riffRaffUpload` SBT task - you'll need to make sure that 
+       the `riffRaffUploadArtifactBucket` and `riffRaffUploadManifestBucket` settings are configured in your SBT file
+ 1. It is recommended to set the `riffRaffManifestProjectName` - this is the name that you type into Riff-Raff when
+    starting a deployment
+    
+Using an autoscaling deployment type
+------------------------------------
+
+The most common deployment uses the `autoscaling` deployment type. This uploads your artifacts to an S3 bucket,
```
Comment (philwills) [inline riff-raff/public/docs/howto/configure-a-project.md]: Might it be better to link the autoscaling deployment generated docs here.

File: riff-raff/public/docs/reference/riff-raff.yaml.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
+The riff-raff.yaml configuration file
+=====================================
+
+Whilst writing riff-raff.yaml files you are strongly advised to use the Validate Template feature in Riff-Raff to get
```
Comment (philwills) [inline riff-raff/public/docs/reference/riff-raff.yaml.md]: Link?

File: riff-raff/public/docs/howto/configure-a-project.md (hunk 2 above)
Comment (sihil) [inline riff-raff/public/docs/howto/configure-a-project.md]: We should probably move these docs into the repo - could probably do with some work. I've tweaked the copy a little, but will leave some of the instructions here for now.

File: riff-raff/public/docs/howto/configure-a-project.md (hunk 3 above)
Comment (sihil) [inline riff-raff/public/docs/howto/configure-a-project.md]: Have added a link as well.

File: riff-raff/public/docs/reference/riff-raff.yaml.md (hunk 4 above)
Comment (sihil) [inline riff-raff/public/docs/reference/riff-raff.yaml.md]: hmm, yeah. links to something outside of the docs, but seems to work.

**Comment by sihil** [comment]
I'm going to merge this. I think it's a much more visible and a better starting point to improve and embellish. I wholeheartedly encourage others to get involved 🐵

**Pull request merged**